sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -339,10 +340,17 @@ private[hive] class HiveClientImpl(
 
   override def getDatabase(dbName: String): CatalogDatabase = withHiveState {
     Option(client.getDatabase(dbName)).map { d =>
+      // default database's location always use the warehouse path,
+      // and since the location of database stored in metastore is qualified,
+      // here we also make qualify for warehouse location
+      val dbLocation = if (dbName == SessionCatalog.DEFAULT_DATABASE) {
+        SessionCatalog.makeQualifiedPath(sparkConf.get(WAREHOUSE_PATH), hadoopConf).toString
```

Comment:
This won't work for `InMemoryCatalog`, isn't it?

You should either implement this logic in all `ExternalCatalog`s, all put it in `SessionCatalog`